passes/elim_dead_code.cpp

```diff
@@ -187,6 +187,18 @@ const IR::Node *ElimDeadCode::preorder(IR::MethodCallStatement *stmt) {
         }
     }
 
+    // Action annotated with `@defaultonly` is ignored before. If no action can be reachable
+    // based on previous analysis, we replace the apply call to the default action call.
+    // Only when the default action has an empty body, we remove the apply.
+    auto *defaultAction = table.getDefaultAction()->to<IR::MethodCallExpression>();
+    if (defaultAction != nullptr) {
+        auto decl = getActionDecl(refMap, *defaultAction);
+        if (decl.has_value() && !decl.value()->body->components.empty()) {
+            printInfo("Replacing table apply with default action %1%", defaultAction);
```

Comment:
Well, we can still add some form of eliminated node here. Otherwise we do not track that we have modified the table here. What do you think?

Reply:
In the sense of `eliminatedNodes.push_back(stmt);` or something similar. Our current ref files do not quite capture this ambiguity.

Reply:
Yes. I think a better way is to create another vector to store these because they are not simply eliminated. What do you think?

Reply:
You could make it a vector of pairs, with the original node and the replaced node.

There is a question on the performance impact when we keep adding nodes. Recall that this code may run on every control-plane update message. But I do not think it will have much impact because we only invoke the elimination when the semantics have changed. Still, it's something we should keep in mind.

Reply:
> You could make it a vector of pairs, with the original node and the replaced node.
That sounds good. I will do it soon.

And yes, I agree on the performance thing.